### PR TITLE
Implementation of a simple BoundingBox

### PR DESCRIPTION
**Pull request by PyMarc2**
A `BoundingBox` is necessary to rapidly know the 3d extent of a structure, wheter it is a polygon, a solid or a scene.

The `BoundingBox` is analogous to the cuboid, it has x, y and z values.
It cannot interpret or nuance for any curve-like object, thus cannot be directly used to know if an object contains another. 

The `BoundingBox` will be used for raytracing/intersections calculations and for the object-search algorithms

### CODE REVIEW

**Review by JLBegin — CHANGES_REQUESTED** [review]
I approve the changes, but I would like an update on the bounding box property tests as discussed before merging.

**Comment by PyMarc2** [comment]
waiting for confirmation

**Review by JLBegin — APPROVED** [review]

**Pull request merged**
